### ISSUE TITLE
Progress bars in implementations of migrate spatialite

### ISSUE THREAD

**Issue by leendertvanwolfswinkel** [nens/threedi-results-analysis#1065]
- Processing algorithm
- Schematisation editor: when you are asked to migrate it, i.e. when adding a schematisation to the project
- Models & simulations: when you are asked to migrate it

Depends on nens/threedi-schema#136

**Comment by leendertvanwolfswinkel** [comment]
nens/threedi-schema#136 is included in `threedi-schema >= 0.228.1`

**Comment by ldebek** [comment]
@Jonasvsl It seems that there might be an issue with `threedi-schema` (0.300.8) which returns `division by zero` error message if I put a callback function to track progress. My gues is that it have something to do with a total steps count calculation inside `threddi-schema`, but it would be good if library devs will check it out as functionality essentially does not work for me. Code I'm using for migrating Sqlite schematisation to the GeoPackage format is like this:
```
threedi_db = ThreediDatabase(schematisation_filepath)
threedi_db.schema.upgrade(backup=False, progress_func=progress_callback)
```

**Comment by Jonasvsl** [comment]
@elisalle can you answer lukasz' question?

**Comment by ldebek** [comment]
@Jonasvsl @elisalle Any news on `progress_func` issue?

**Comment by Jonasvsl** [comment]
@ldebek no, we were busy with the backend release of this morning. I'll ask the team

**Comment by elisalle** [comment]
@ldebek this is probably going wrong on this line:
https://github.com/nens/threedi-schema/blob/038ea0e05f4dbb0ec81ac4fe427fb2fe2a63cb25/threedi_schema/application/upgrade_utils.py#L23
I think the assumption was that there would always be at least one step, the step representing the action taken; with no steps to be made, there's not much progress to show.

**Comment by margrietpalm** [comment]
@ldebek This should be fixed in threedi-schema 0.300.17 (and higher)

**Comment by elisalle** [comment]
It's now also updated on the schema_300 branch.

**Comment by leendertvanwolfswinkel** [comment]
I tested this and I think it still needs some improvement. What I see is a progress bar that (relatively quickly) goes up to 100%, then goes back to 0% and stays there for quite some time (I suppose this is the "convert to geopackage" part?), often longer than the "going from 0 to 100%" part. Improvements that I suggest:
- More specific messages (i.e. "Migrating to schema version 223...", "Converting Spatialite to Geopackage...", etc.)
- Include spatialite to geopackage conversion in the calculation of the progress percentage

@margrietpalm I think this mainly requires changes in threedi-schema. Could you please take this up?

**Comment by margrietpalm** [comment]
@leendertvanwolfswinkel I don't fully understand the problem. I think the only way the bar can go back to 0 is when the upgrade function runs again. Without the exact steps and my limited knowledge of how the plugins work, I cannot determine what is going on here.

**Comment by leendertvanwolfswinkel** [comment]
@margrietpalm I deleted my previous comment because apparently I did something wrong when installing the new plugin versions. It works correctly now

**Comment by leendertvanwolfswinkel** [comment]
@margrietpalm I got the following error when running the schematisation checker on a not up-to-date schematisation from within the upload wizard: `setValue(self, value: int): argument 1 has unexpected type 'float'`

How to reproduce: 
- In the 3Di Model and Simulations panel, choose a schematisation that does not have the current schema version. Do not add it to the project if prompted. 
- Click Upload 
- Click next until you get to the Check schematisation page
- Click Check schematisation
- Error appears

**Comment by leendertvanwolfswinkel** [comment]
@margrietpalm @ldebek I think I found and fixed the issue. <img width="798" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBeEJ0Q1E9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--1c3fe655355bfbd32f3a118dda26b42b57c884bf/image.png" alt="image.png" />

The reason that this issue also arises when using functionality from plugins like 3Di Schematisation Editor and 3Di Models & Simulations seems to be that they also have a UICommunication class, but it uses the UICommunication object from the 3Di Models & Simulations plugin when all plugins are loaded at the same time. 

Perhaps it would be a good idea to move this to `threedi-mi-utils`?
